src/main/java/htsjdk/tribble/index/AbstractIndex.java

```diff
@@ -67,7 +70,7 @@
     private final static long NO_TS = -1L;
 
     protected int version;                    // Our version value
-    protected File indexedFile = null;         // The file we've created this index for
+    protected Path indexedPath = null;         // The file we've created this index for
```

Comment:
Yeah, it's tricky.  Anything field that's visible to a subclass of a non-final class is technically a breaking change since someone could be relying on it if they implement a subclass.  It's why I'm so strongly in favor of making classes final and of making variable private with accessors if they need to be accessed.